src/main/java/com/commercetools/sync/producttypes/helpers/BatchProcessor.java

```diff
@@ -0,0 +1,154 @@
+package com.commercetools.sync.producttypes.helpers;
+
+import com.commercetools.sync.commons.exceptions.InvalidProductTypeDraftException;
+import com.commercetools.sync.commons.exceptions.InvalidReferenceException;
+import com.commercetools.sync.producttypes.ProductTypeSync;
+import io.sphere.sdk.products.attributes.AttributeDefinitionDraft;
+import io.sphere.sdk.products.attributes.AttributeType;
+import io.sphere.sdk.products.attributes.NestedAttributeType;
+import io.sphere.sdk.products.attributes.SetAttributeType;
+import io.sphere.sdk.producttypes.ProductTypeDraft;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.commercetools.sync.commons.helpers.BaseReferenceResolver.BLANK_ID_VALUE_ON_RESOURCE_IDENTIFIER;
+import static java.lang.String.format;
+import static java.util.Collections.emptySet;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+
+public class BatchProcessor {
+    static final String PRODUCT_TYPE_DRAFT_KEY_NOT_SET = "ProductTypeDraft with name: %s doesn't have a key. "
```

Comment:
why? Isn't this more helpful to the user to see right away which productType doesn't have a key to easily fix it? instead of having to go through all productTypes and find which one has a key and which not.

Reply:
no, I mean `ProductTypeDraft with name: %s doesn't have a key.`  is necessary and related to the product type but ` Please make sure all productType drafts have keys.` part of the message is not related to that product type, it is related to the all attributes.

Reply:
true, but where else would I add it as a warning message. It doesn't hurt to have it here too. It shows that not only this `productTypeDraft` should have a key but all of them.